etcdserver/apply_auth.go

```diff
@@ -158,6 +174,22 @@ func (aa *authApplierV3) Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error) {
 	return aa.applierV3.Txn(rt)
 }
 
+func (aa *authApplierV3) LeaseRevoke(lc *pb.LeaseRevokeRequest) (*pb.LeaseRevokeResponse, error) {
+	if !lc.Expire {
+		// If lc.Expire == true, it means LeaseRevoke() is invoked by time progress,
+		// no permission check is required.
+		lease := aa.lessor.Lookup(lease.LeaseID(lc.ID))
```

Comment:
dedup this code into `if err := aa.checkLeasePuts(leaseId); err != nil { return nil, err }`?